aten/src/ATen/core/ATenDispatch.h

```diff
@@ -128,9 +129,19 @@ class CAFFE2_API ATenOpTable {
 
   C10_NORETURN void reportError(TensorTypeId tid) const;
 
+  const FunctionSchema& function_schema() const {
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
It's a bit shameful we have to take out the lock every time we do an operator call.

Reply:
(once flag would be more efficient here.)